cpp/src/decisiontree/batched-levelalgo/metrics.cuh

```diff
@@ -96,101 +145,146 @@ DI void entropyGain(int* shist, DataT* sbins, Split<DataT, IdxT>& sp, IdxT col,
       nLeft += shist[i * 2 * nclasses + j];
     }
     auto nRight = len - nLeft;
-    auto invLeft = One / nLeft;
-    auto invRight = One / nRight;
     auto gain = DataT(0.0);
-    for (IdxT j = 0; j < nclasses; ++j) {
-      int val_i = 0;
-      if (nLeft != 0) {
+    // if there aren't enough samples in this split, don't bother!
+    if (nLeft < min_samples_leaf || nRight < min_samples_leaf) {
+      gain = -NumericLimits<DataT>::kMax;
+    } else {
+      auto invLeft = One / nLeft;
+      auto invRight = One / nRight;
+      for (IdxT j = 0; j < nclasses; ++j) {
+        int val_i = 0;
         auto lval_i = shist[i * 2 * nclasses + j];
         if (lval_i != 0) {
           auto lval = DataT(lval_i);
           gain += raft::myLog(lval * invLeft) * lval * invlen;
```

Comment:
Unrelated to this particular change, but while we are at repairing the metrics, we might as well do it. The `myLog` here is to the base `e`. It should be to the base 2. Simply multiplying by `1/log(2)` would suffice here.

Reply:
Done